### ISSUE TITLE
Display success notification in main dashboard after Ad Blocking Recovery tag set up

### ISSUE THREAD

**Issue by kuasha420** [google/site-kit-wp#6957]
## Feature Description

There should be a success notification indicating that the Ad Blocking Recovery has been setup successfully and instructing user to make sure the message is created in AdSense as seem in the [Figma](https://www.figma.com/file/9jzqdDjP7pCL8MiYHtxxoJ/Ad-Blocker-Detection-2?node-id=766-3148&t=zbt6T8bNFSmrSWTI-0) design. 

![Screenshot_20230422_002429](https://user-images.githubusercontent.com/6187401/233709503-a0a0922e-53f9-4598-9dc2-1b2300554e39.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A banner notification should appear on the Site Kit main dashboard after Ad Blocking Recovery is successfully set up.
* This banner should contain the contents in the above screenshot/Figma design
* The banner should have a single CTA (`"OK, Got it!"`) that, when clicked, permanently dismisses the notification
* The notification should appear below the `ZeroDataStateNotifications`, in terms of priority.

## Implementation Brief

* Once #6960 is merged.
* Create a new component, `AdBlockingRecoveryNotification` in `assets/js/components/notifications` with the following:
  * Check the `adBlockingRecoverySetupStatus` from the `getSettings` selector in the `modules/adsense` datastore.
  * Return `null` if the `adBlockingRecoverySetupStatus` is NOT equal to `setup-confirmed` OR the notification has been dismissed using the `isItemDismissed` selector from the `core/user` store.`isItemDismissed` selector from the `core/user` store.
  * Otherwise, render the `BannerNotification` component with the following props:
    * `id`: `ad-blocking-recovery-setup-success`
    * `title`: Translatable `"You successfully added the ad blocking recovery tag"`.
    * `description`: Translatable `"Make sure to also create the message in AdSense, otherwise this feature won't work."`.
    * `ctaLabel`: Translatable "`OK, Got it!`".
    * `onCTAClick`: Pass a callback function that dispatches the `dismissItem` action of the `core/user` store and pass the notification ID (`ad-blocking-recovery-setup-success`) to it.
    * `type`: `"win-success"`.
    * `WinImageSVG`: Export the following SVG from the [Figma](https://www.figma.com/file/9jzqdDjP7pCL8MiYHtxxoJ/Ad-Blocker-Detection-2?node-id=766-3148&t=zbt6T8bNFSmrSWTI-0) design, and pass it as an inline function component (e.g., `() => <SVGImageComponent />`).
    
![ad-blocking-recovery-svg](https://github.com/google/site-kit-wp/assets/24408261/acc457e1-5555-47ab-9a87-da3ede12c008)
    
* Render the `AdBlockingRecoveryNotification` component within `BannerNotifications` just below the `ZeroDataStateNotifications` banner only if the `adBlockerDetection` feature flag is enabled.
* Add storybook stories for the `AdBlockingRecoveryNotification` notification banner.

### Test Coverage

* Add some basic unit tests for the `AdBlockingRecoveryNotification` component.

## QA Brief

- Since the ad blocking recovery onboarding hasn't been implemented yet, it isn't possible to test this in a general flow.
- In the Site Kit main dashboard, run the following code in the console of your browser developer tools:
`googlesitekit.data.dispatch( 'modules/adsense' ).setAdBlockingRecoverySetupStatus( 'setup-confirmed' )`
- This will make Site Kit assume that ad blocking recovery setup is successful and show the notification banner.
- Verify that the banner looks and works according to the ACs.
- Verify the `AdBlockingRecoveryNotification` Storybook story.

## Changelog entry

* Display a success notification on the Main Dashboard after completing the Ad Blocking Recovery tag setup.

**Comment by tofumatt** [comment]
Just needed to tweak the translatable string, as the ACs explicitly called for `'OK, Got it!'` rather than the text that's in the Figma design, because it's the same string we use elsewhere in the plugin:

<img width="306" alt="CleanShot 2023-05-12 at 14 45 47" src="https://github.com/google/site-kit-wp/assets/90871/c0417bfb-79be-4844-ba1b-3decb8293c7a">

(It doesn't make sense to have two very similar strings like this from a localisation/UX perspective 😅)

IB ✅

**Comment by nfmohit** [comment]
@kuasha420 @tofumatt Should the word `AdSense` in the description link to anything as shown in the designs? If so, to where? Thanks!

**Comment by kuasha420** [comment]
We can link it to the Privacy & Messaging section of AdSense, ie. `https://www.google.com/adsense/new/u/0/pub-XXX/privacymessaging/ad_blocking` with proper pub ID.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@arafat, on desktop, the width and height of the svg is considerably larger than on the [figma](https://www.figma.com/file/9jzqdDjP7pCL8MiYHtxxoJ/Ad-Blocker-Detection-2?type=design&node-id=766-3148&t=26bjaSBLAdIPCT8i-0) design. 

![image](https://github.com/google/site-kit-wp/assets/73545194/5820355d-71fa-476e-bc12-365548695dba)

**Verified:**

- A banner notification appears on the Site Kit main dashboard after Ad Blocking Recovery is successfully set up.
  - This banner contains the contents in the above screenshot/Figma design.
  - The banner has a single CTA ("OK, Got it!") that, when clicked, permanently dismisses the notification
  - The notification appears below the ZeroDataStateNotifications, in terms of priority.

Ran through some additional tests: 
- Checked to make sure that the banner did not appear on the entity dashboard
- Checked to make sure that the banner did not appear on the view only dashboard. 
- Checked the banner on mobile to ensure the graphic does not appear.  It doesn't mention this in the AC, but this is the behaviour for other notifications. 

https://github.com/google/site-kit-wp/assets/73545194/b5b8238a-b986-45cf-9932-99ee3e4ece0e

**Comment by nfmohit** [comment]
Good catch, @wpdarren!

I think this is the same composition as most other banner notifications, but the size is prominently visible here due to less text.

I have gone ahead and made a follow-up PR #7109 and set a maximum width to the graphic so that it isn't larger than expected.

Thanks!
